lib/jekyll/document.rb

```diff
@@ -258,9 +258,11 @@ def read(opts = {})
           read_post_data
         rescue SyntaxError => e
           Jekyll.logger.error "Error:", "YAML Exception reading #{path}: #{e.message}"
+          raise e if self.site.config["strict_front_matter"]
```

Comment:
`self` is not required here and in the line further below.
```diff
- raise e if self.site.config["strict_front_matter"]
+ raise e if site.config["strict_front_matter"]
```

I also think multiple `raise`s can be avoided by using the `or` operator:
```ruby
    raise e if (e.is_a? Jekyll::Errors::FatalException || site.config["strict_front_matter"])
    Jekyll.logger.error "Error:", "could not read file #{path}: #{e.message}"
```

/cc @pathawks